lib/iris/_lazy_data.py

```diff
@@ -92,61 +102,93 @@ def _optimum_chunksize(chunks, shape, limit=None, dtype=np.dtype("f4")):
 
     point_size_limit = limit / dtype.itemsize
 
-    # Create result chunks, starting with a copy of the input.
-    result = list(chunks)
-
-    if np.prod(result) < point_size_limit:
-        # If size is less than maximum, expand the chunks, multiplying later
-        # (i.e. inner) dims first.
-        i_expand = len(shape) - 1
-        while np.prod(result) < point_size_limit and i_expand >= 0:
-            factor = np.floor(point_size_limit * 1.0 / np.prod(result))
-            new_dim = result[i_expand] * int(factor)
-            if new_dim >= shape[i_expand]:
-                # Clip to dim size : chunk dims must not exceed the full shape.
-                new_dim = shape[i_expand]
-            else:
-                # 'new_dim' is less than the relevant dim of 'shape' -- but it
-                # is also the largest possible multiple of the input-chunks,
-                # within the size limit.
-                # So : 'i_expand' is the outer (last) dimension over which we
-                # will multiply the input chunks, and 'new_dim' is a value that
-                # ensures the fewest possible chunks within that dim.
-
-                # Now replace 'new_dim' with the value **closest to equal-size
-                # chunks**, for the same (minimum) number of chunks.
-                # More-equal chunks are practically better.
-                # E.G. : "divide 8 into multiples of 2, with a limit of 7",
-                # produces new_dim=6, which would mean chunks of sizes (6, 2).
-                # But (4, 4) is clearly better for memory and time cost.
-
-                # Calculate how many (expanded) chunks fit into this dimension.
-                dim_chunks = np.ceil(shape[i_expand] * 1.0 / new_dim)
-                # Get "ideal" (equal) size for that many chunks.
-                ideal_equal_chunk_size = shape[i_expand] / dim_chunks
-                # Use the nearest whole multiple of input chunks >= ideal.
-                new_dim = int(
-                    result[i_expand]
-                    * np.ceil(ideal_equal_chunk_size / result[i_expand])
-                )
-
-            result[i_expand] = new_dim
-            i_expand -= 1
+    if dims_fixed is not None:
+        if not np.any(dims_fixed):
+            dims_fixed = None
+
+    if dims_fixed is None:
+        # Get initial result chunks, starting with a copy of the input.
+        working = list(chunks)
+    else:
+        # Adjust the operation to ignore the 'fixed' dims.
+        # (We reconstruct the original later, before return).
+        chunks = np.array(chunks)
+        dims_fixed_arr = np.array(dims_fixed)
+        # Reduce the target size by the fixed size of all the 'fixed' dims.
+        point_size_limit = point_size_limit // np.prod(chunks[dims_fixed_arr])
```

Comment:
It would be nice to somehow allow Dask's [special chunk values](https://docs.dask.org/en/stable/array-chunks.html#automatic-chunking) (`-1`, `'auto'` and `None`), but I guess this is at odds with doing this logic on Iris side. Have you tried to inquire if Dask would be interested to add/extend chunking logic to support such use cases? I think it would make the most sense to boot this out of Iris as per the note [here](https://github.com/SciTools/iris/blob/1d5eb3eaf9d183af4e1849e82798bedf876ab53e/lib/iris/_lazy_data.py#L77-L78).

Reply:
### (background, not really an answer!)

The problem with the existing Dask built-in strategy is that is treats all dimensions as equal and tries to even out the chunking across all dims.  Which is pretty clearly unsuitable in most of our cases, given the known contiguity factors in file storage, and especially when the file contains 2d-fields which cannot be efficiently sub-indexed.
Which is why we provided the Iris version, to prioritise splitting of the "outer" dims and keeping the "inner" ones together.

**But it's certainly true that we could feedback and engage, as-per the note you mention.**
Unfortunately though, the dask code is really complicated and engagement seems a bit of an uphill struggle IMHO.
By which I mean ... I have repeatedly attempted to engage with the dask codebase, and maybe contribute, but I've found it very hard to understand, with a lot of unexplained terminology and framework concepts.  
So it seems to me that there is a problem with contributing unless you are an "expert", because there is a lot that you need to know beyond what the "ordinary" user sees.

In fact, as a very relevant example, I considered addressing this same chunking control problem by picking apart the graph of a loaded cube to identify and adjust (or replace) the `iris.fileformats.netcdf.NetCDFDataProxy` objects which are in there.  It's perfectly do-able, but you need to understand the structure of the dask graph and employ various assumptions about it :  However, there is _no public account_ of that, or any apparent intention for "ordinary users" to manipulate them in this way.  So it would seem that a dask graph is really an "opaque" object, and the internal structure might get refactored and change completely, in which case such a solution, being outside intended usage, would simply need re-writing.  So effectively, AFAICT only a dask "expert" is expected to work at that level and, crucially, such a solution only really makes sense _if the code lives in the dask codebase_.

Reply:
> @TomekTrzeciak It would be nice to somehow allow Dask's [special chunk values](https://docs.dask.org/en/stable/array-chunks.html#automatic-chunking) (-1, 'auto' and None),

Reminder of how that is described :
```
Chunks also includes three special values:
   * -1: no chunking along this dimension
   * None: no change to the chunking along this dimension (useful for rechunk)
   * "auto": allow the chunking in this dimension to accommodate ideal chunk sizes
```


From a technical POV, I guess we could respect -1, but the others are a little tricky :  
"Auto" is sort-of what we are already doing, with the dims we don't have an explicit control for.
But it isn't quite what Dask means my it since we will _still_ want to apply a dimension ordering.
**"None"** I think could only really apply if we treat chunking specified on a file variable as the initial setting.  At present, we do use that as a starting point anyway, so I'm not sure how this would differ from "auto" in this usage.

Reply:
> From a technical POV, I guess we could respect -1, but the others are a little tricky :
> **"Auto"** is sort-of what we are already doing, with the dims that _don't_ have an explicit control given. But it isn't quite what Dask means, since we will still want to apply our dimension ordering priority.
> **"None"** I think could only really apply if we treat chunking specified on a file variable as the initial setting. At present, we do use that as a starting point anyway, so I'm not sure how this would differ from "auto" in this usage.
> 
> **Can you suggest what the interpretations would be -- how would this work ?**

For **None** the only sensible interpretation I can think of is to keep the chunking from the netcdf file itself as you suggest, but dunno what Dask is doing in that case.

I reckon **Auto** would require repeating the logic that Dask is doing but restricted only to the specified dimensions. Seem quite tricky, probably best not to support it if the implementation is going to live on the Iris' side.

Reply:
> > **"None"** I think could only really apply if we treat chunking specified on a file variable as the initial setting. At present, we do use that as a starting point anyway, so I'm not sure how this would differ from "auto" in this usage.
> 
> For **None** the only sensible interpretation I can think of is to keep the chunking from the netcdf file itself as you suggest, but dunno what Dask is doing in that case.

Sorry, I've re-read your comment and I think this is not what you suggest. My interpretation of **None** is that it behaves the same as if you would explicitly specify chunks equal to those already existing on the variable, i.e., this is their final rather than initial value.